Scripts/RemediationScripts/Remediate-SetClusterProtectionLevelForServiceFabric.ps1

```diff
@@ -216,28 +232,94 @@ function Set-ClusterProtectionLeveltoEncryptandSignforServiceFabric
+            Write-Host $([Constants]::SingleDashLine)	
+            break	
+        }
+
+         Write-Host "Fetching all Service Fabric(s) failing for the [$($controlIds)] control from [$($Path)]..." -ForegroundColor $([Constants]::MessageType.Info)
+         Write-Host $([Constants]::SingleDashLine)
+         $controlForRemediation = Get-content -path $Path | ConvertFrom-Json
+         $controls = $controlForRemediation.ControlRemediationList
+         $resourceDetails = $controls | Where-Object { $controlIds -eq $_.ControlId };
+
+          $validResources = $resourceDetails.FailedResourceList | Where-Object {![String]::IsNullOrWhiteSpace($_.ResourceId)}	
+        if(($resourceDetails | Measure-Object).Count -eq 0 -or ($validResources | Measure-Object).Count -eq 0)
+        {
+            	
+            Write-Host "No Service Fabric(s) found in input json file for remediation." -ForegroundColor $([Constants]::MessageType.Error)	
+            Write-Host $([Constants]::SingleDashLine)	
+            break	
+        
+        }
+         $validResources | ForEach-Object { 	
+            try	
+            {
+            $ServiceFabricResource =  Get-AzServiceFabricCluster -ResourceGroupName $_.ResourceGroupName -Name $_.ResourceName -ErrorAction SilentlyContinue
+            $ServiceFabricDetails += $ServiceFabricResource  | Select-Object @{N='ResourceId';E={$_.Id}},
+                                                                          @{N='ResourceGroupName';E={$_.Id.Split("/")[4]}},
+                                                                          @{N='ResourceName';E={$_.Name}}, 
+                                                      @{N='IsClusterProtectionLevelExist';E={$_.FabricSettings.Parameters.Name -Contains 'ClusterProtectionLevel'}},
+                                                                          @{N='ClusterProtectionLevelValue';E={if(($_.FabricSettings.Parameters.Name -Contains 'ClusterProtectionLevel') -eq $true)
+                                                                          { 
+                                                                                $param = $_.FabricSettings | Where-Object {$_.Parameters.Name -eq 'ClusterProtectionLevel'}
+                                                                                $param.Parameters.Value
+                                                                          }                                                                          
+                                                                          }},
+                                                                          @{N='NodeVMCount';E={$_.NodeTypes.VmInstanceCount}}
+             }
+            catch
+            {
+                Write-Host "Valid resource id(s) not found in input json file. ErrorMessage [$($_)]" -ForegroundColor $([Constants]::MessageType.Error)	
+                Write-Host "WARNING: Skipping the Resource [$($_.ResourceName)]..."
```

Comment:
Remove 'WARNING' test and add ':' after resource

Reply:
This is done, thanks